test/unit/BFactory.tree

```diff
@@ -5,15 +5,12 @@ BFactoryTest::constructor
 BFactoryTest::newBPool
 └── when called
     ├── it should call _newBPool
+    ├── it should deploy a new BPool
     ├── it should add the newBPool to the mapping of pools
     ├── it should return the address of the new BPool
     ├── it should emit a PoolCreated event
     └── it should set the controller of the new BPool to the caller
 
-BFactoryTest::_newBPool
-└── when called
-    └── it should deploy a new BPool
-
```

Comment:
i don't follow why we're doing this, isn't it better to test that the internal gets called and test the internal separately?